app/uk/gov/hmrc/gatekeeper/controllers/EmailsPreferencesController.scala

```diff
@@ -115,47 +115,57 @@ class EmailsPreferencesController @Inject() (
       }
     }
 
-  def addAnotherTaxRegimeOption(selectOption: String, selectedCategories: Option[List[String]], selectedTopic: Option[String]): Action[AnyContent] =
+  def addAnotherTaxRegimeOption(selectOption: String, selectedCategories: Option[List[String]], offset: Int = 0, limit: Int = 15): Action[AnyContent] =
     anyStrideUserAction { implicit request =>
       selectOption.toUpperCase match {
         case "YES" => Future.successful(Redirect(routes.EmailsPreferencesController.selectTaxRegime(selectedCategories)))
-        case _     =>
-          for {
-            categories                <- apiDefinitionService.apiCategories
-            selectedApiCategories      = filterSelectedApiCategories(selectedCategories, categories)
-            users                     <- developerService.fetchDevelopersBySpecificTaxRegimesEmailPreferences(selectedApiCategories)
-            selectedApiCategoryDetails = filterSelectedCategories(selectedCategories, categories)
-            usersAsJson                = Json.toJson(users)
-          } yield Ok(emailPreferencesSelectedUserTaxRegimeView(
-            users,
-            usersAsJson,
-            usersToEmailCopyText(users),
-            selectedApiCategoryDetails
-          ))
+        case _     => Future.successful(Redirect(routes.EmailsPreferencesController.selectedUserTaxRegime(selectedCategories, offset, limit)))
       }
     }
 
+  def selectedUserTaxRegime(selectedCategories: Option[List[String]], offset: Int, limit: Int): Action[AnyContent] = {
+    anyStrideUserAction { implicit request =>
+      for {
+        categories                <- apiDefinitionService.apiCategories
+        selectedApiCategories      = filterSelectedApiCategories(selectedCategories, categories)
+        userPaginatedResponse     <- developerService.fetchDevelopersBySpecificTaxRegimesEmailPreferencesPaginated(selectedApiCategories, offset, limit)
+        totalCount                 = userPaginatedResponse.totalCount
+        users                      = userPaginatedResponse.users.filter(_.verified)
+        selectedApiCategoryDetails = filterSelectedCategories(selectedCategories, categories)
+        usersAsJson                = Json.toJson(users)
+      } yield Ok(emailPreferencesSelectedUserTaxRegimeView(
+        users,
+        usersAsJson,
+        usersToEmailCopyText(users),
+        selectedApiCategoryDetails,
+        offset,
+        limit,
+        totalCount
+      ))
+    }
+  }
+
   def selectUserTopicPage(selectedTopic: Option[String]): Action[AnyContent] = anyStrideUserAction { implicit request =>
     Future.successful(Ok(emailPreferencesSelectUserTopicView(selectedTopic.map(TopicOptionChoice.withName))))
   }
 
-  def selectedUserTopic(selectedTopic: Option[String]): Action[AnyContent] =
+  def selectedUserTopic(selectedTopic: Option[String], offset: Int = 0, limit: Int = 15): Action[AnyContent] =
```

Comment:
I think the `15` here and multiple other places should be a configuration, otherwise it'll be really fiddly to change later.

Reply:
Addressed